code/modules/mob/living/carbon/xenomorph/castes/hivelord/abilities_hivelord.dm

```diff
@@ -260,8 +260,155 @@ GLOBAL_LIST_INIT(thickenable_resin, typecacheof(list(
+	name = "Healing Infusion"
+	action_icon_state = "heal_xeno"
+	mechanics_text = "Psychically infuses a friendly xeno with regenerative energies, greatly improving its natural healing."
+	cooldown_timer = 5 SECONDS
+	plasma_cost = 200
+	keybind_signal = COMSIG_XENOABILITY_HEALING_INFUSION
+	var/heal_range = HIVELORD_HEAL_RANGE
+	var/healing_infusion_aura
+
+
+/datum/action/xeno_action/activable/healing_infusion/can_use_ability(atom/target, silent = FALSE, override_flags)
+	. = ..()
+	if(!.)
+		return FALSE
+	if(QDELETED(target))
+		return FALSE
+
+	if(!isxeno(target))
+		if(!silent)
+			to_chat(owner, "<span class='warning'>We can only target fellow sisters with [src]!</span>")
+		return FALSE
+
+	if(!check_distance(target, silent))
+		return FALSE
+
+	var/mob/living/carbon/xenomorph/patient = target
+	if(patient.has_healing_infusion)
+		if(!silent)
+			to_chat(owner, "<span class='warning'>[patient] is already benefitting from our [src]!</span>")
+		return FALSE
+
+	if(!CHECK_BITFIELD(use_state_flags|override_flags, XACT_IGNORE_DEAD_TARGET) && patient.stat == DEAD)
+		if(!silent)
+			to_chat(owner, "<span class='warning'>It's too late. This sister won't be coming back.</span>")
+		return FALSE
+
+/datum/action/xeno_action/activable/healing_infusion/proc/check_distance(atom/target, silent)
+	var/dist = get_dist(owner, target)
+	if(dist > heal_range)
+		if(!silent)
+			to_chat(owner, "<span class='warning'>Too far for our reach... We need to be [dist - heal_range] steps closer!</span>")
+		return FALSE
+	else if(!owner.line_of_sight(target))
+		if(!silent)
+			to_chat(owner, "<span class='warning'>We can't focus properly without a clear line of sight!</span>")
+		return FALSE
+	return TRUE
+
+
+/datum/action/xeno_action/activable/healing_infusion/use_ability(atom/target)
+	if(owner.action_busy)
+		return FALSE
+
+	owner.face_atom(target) //Face the target so we don't look stupid
+
+	if(!do_mob(owner, target, 0.5 SECONDS, BUSY_ICON_FRIENDLY, BUSY_ICON_MEDICAL))
+		return FALSE
+
+	owner.visible_message("<span class='xenodanger'>\the [owner] infuses [target] with mysterious energy!</span>", \
+	"<span class='xenodanger'>We empower [target] with our [src]!</span>")
+
+	playsound(target, 'sound/effects/magic.ogg', 25) //Cool SFX
+	playsound(owner, 'sound/effects/magic.ogg', 25) //Cool SFX
+	owner.beam(target,"medbeam",'icons/effects/beam.dmi',10, 10,/obj/effect/ebeam,1)
+	new /obj/effect/temp_visual/telekinesis(get_turf(owner))
+	new /obj/effect/temp_visual/telekinesis(get_turf(target))
+	to_chat(target, "<span class='xenodanger'>Our wounds begin to knit and heal rapidly as [owner]'s healing energies infuse us.</span>") //Let the target know.
+
+	var/mob/living/carbon/xenomorph/patient = target
+
+	patient.has_healing_infusion = TRUE
+	healing_infusion_aura = filter(type = "outline", size = 1, color = COLOR_VERY_PALE_LIME_GREEN) //Set our cool aura; also confirmation we have the buff
+	patient.filters += healing_infusion_aura
```

Comment:
All these filter related functions should use the correct helpers @Hughgent merge them already pls